mne/preprocessing/tests/test_ica.py

```diff
@@ -608,4 +610,58 @@ def test_fit_params():
     ICA(fit_params=fit_params)  # test no side effects
     assert_equal(fit_params, {})
 
+
+@requires_sklearn
+def test_bad_channels():
+    """Test if exception is raised when unsupported channels are used.
+    Test for raw and evoked data"""
+    chs = [i for i in _kind_dict]
+    data_chs = _DATA_CH_TYPES_SPLIT + ['eog']
+    chs_bad = list(set(chs) - set(data_chs))
+    info = create_info(len(chs), 500, chs)
+    data = np.random.rand(len(chs), 50)
+    raw = RawArray(data, info)
+    data = np.random.rand(100, len(chs), 50)
+    epochs = EpochsArray(data, info)
+
+    n_components = 0.9
+    ica = ICA(n_components=n_components)
+
+    for inst in [raw, epochs]:
+        for ch in chs_bad:
+            # Test case for only bad channels
+            picks_bad1 = pick_types(inst.info, exclude='bads',
+                                    **{str(ch): True})
+            # Test case for good and bad channels
+            picks_bad2 = pick_types(inst.info, exclude='bads',
+                                    **{str(ch): True})
```

Comment:
Or you can always add some "bad channels" (i.e., in reality fine but useful to mark as bad for testing) to `inst.info['bads']` if it's worth tetsing the functionality